baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/PreferAssertjTests.java

```diff
@@ -988,63 +841,335 @@ public void fix_assertThatIntDescription() {
                         "Test.java",
                         "import static org.assertj.core.api.Assertions.assertThat;",
                         "import static org.hamcrest.Matchers.is;",
-                        "",
-                        "import org.assertj.core.api.HamcrestCondition;",
                         "class Test {",
                         "  void foo(int value) {",
-                        "    assertThat(value).describedAs(\"desc\").is(new HamcrestCondition<>(is(1)));",
+                        "    assertThat(value).describedAs(\"desc\").isEqualTo(1);",
                         "  }",
                         "}")
                 .doTest(BugCheckerRefactoringTestHelper.TestMode.TEXT_MATCH);
     }
 
     @Test
-    public void fix_matcherAssertThatString() {
+    public void fix_matcherAssertThatString_startsWith() {
         test()
                 .addInputLines(
                         "Test.java",
                         "import static org.hamcrest.MatcherAssert.assertThat;",
-                        "import static org.hamcrest.Matchers.is;",
+                        "import static org.hamcrest.Matchers.startsWith;",
                         "class Test {",
                         "  void foo(String value) {",
-                        "    assertThat(value, is(\"str\"));",
+                        "    assertThat(value, startsWith(\"str\"));",
                         "  }",
                         "}")
                 .addOutputLines(
                         "Test.java",
                         "import static org.assertj.core.api.Assertions.assertThat;",
-                        "import static org.hamcrest.Matchers.is;",
+                        "import static org.hamcrest.Matchers.startsWith;",
                         "",
                         "import org.assertj.core.api.HamcrestCondition;",
                         "class Test {",
                         "  void foo(String value) {",
-                        "    assertThat(value).is(new HamcrestCondition<>(is(\"str\")));",
+                        "    assertThat(value).is(new HamcrestCondition<>(startsWith(\"str\")));",
```

Comment:
It would be great to automatically handle startsWith and contains. i.e. `assertThat(value).startsWith("str")`

Reply:
Orrr we could just do a refaster layer that will turn these hamcresty things into idomatic assertJ?

Reply:
If it works, however there are a few places where the HamcrestCondition results in code that doesn't compile: https://github.com/joel-costigliola/assertj-core/issues/1607

Reply:
Updated to support startsWith, endsWIth, and contains.